server/sched.go

```diff
@@ -166,6 +167,17 @@ func (s *Scheduler) processPending(ctx context.Context) {
 						gpus = s.getGpuFn()
 					}
 
+					if pending.origNumCtx == -1 {
+						if len(gpus) == 1 && gpus[0].Library != "cpu" && gpus[0].TotalMemory <= 4096*1024*1024 {
+							slog.Info("GPU is small, limiting context window", "num_ctx", smallGpuContextLength)
```

Comment:
'nit - maybe "GPU is small, limiting default context window" - so it's clearer you can override it to trade some CPU layers for larger context.

Reply:
I like that! Will change it